lib/next_ls.ex

```diff
@@ -171,6 +173,66 @@ defmodule NextLS do
     {:reply, symbols, lsp}
   end
 
+  # TODO extract to a separate module
+  # TODO make this work for modules and structs
+  # TODO how to handle aliases?
+  # TODO can we do this in a single DB query?
+  def handle_request(%TextDocumentReferences{params: %{position: position, text_document: %{uri: uri}}} = request, lsp) do
+    import NextLS.DB.Query
+
+    file = URI.parse(uri).path
+    line = position.line + 1
+    col = position.character + 1
+
+    result =
+      dispatch(lsp.assigns.registry, :databases, fn databases ->
+        for {database_pid, _} <- databases do
+          query = ~Q"""
+          SELECT identifier, arity, type, module
+          FROM "references" as refs
+          WHERE refs.file = ?
+            AND refs.start_line <= ? AND refs.end_line >= ?
+            AND refs.start_column <= ? AND refs.end_column >= ?
+          ORDER BY refs.id ASC
+          LIMIT 1;
+          """
+
+          [[identifier, arity, type, module]] = DB.query(database_pid, query, [file, line, line, col, col])
```

Comment:
in case you trigger find references on something that has no reference (like whitespace or somewhere weird), you'll want to account for this returning an empty list.

Reply:
This should be handled with the latest commit.